tle: add `-m` metadata flag

alienx5499 opened the pull request:
### Summary
Adds a metadata mode to `tle`:
- `tle -m` prints drand network metadata (chain hash, current round, public key, scheme) in YAML.
- `tle -m [INPUT]` prints ciphertext metadata by parsing the tlock stanza (round, chain hash) and estimating the corresponding time via round-to-time inversion.

### Changes
- CLI
  - Update usage to document `--metadata`/`-m` and examples.
  - When `-m` is used with an input file, parse armored age header to extract the `tlock` stanza.
- Implementation
  - New `commands/metadata.go` that:
    - Parses armored header and decodes base64 lines to find `-> tlock <round> <chainhash>`.
    - Estimates the timestamp for the round using a bounded binary search over `network.RoundNumber`.
  - Wire `cmd/tle/tle.go` to call network metadata or ciphertext metadata depending on presence of `INPUT`.

### Usage
```bash
# Network metadata
tle -m

# Ciphertext metadata (armored .age/.tle)
tle -m path/to/encrypted.age
```

### Notes
- Ciphertext metadata currently supports armored age files.
- Time is an estimate computed from the network period/genesis via `RoundNumber` inversion.

### Motivation
Addresses `Add a -m flag to tle to see metadata` and Closes #30.

Closes #30

AnomalRoil commented on cmd/tle/commands/metadata.go:
```diff
@@ -0,0 +1,136 @@
+	// Only support armored input for metadata extraction in this change.
+	if start, _ := rr.Peek(len(armor.Header)); string(start) != armor.Header {
+		return fmt.Errorf("metadata from INPUT currently supports only armored age files")
+	}
```
Why not use an https://pkg.go.dev/filippo.io/age@v1.2.1/armor reader directly instead of having to handling armoring manually?

alienx5499 commented:
Hi @AnomalRoil,

I've addressed your feedback. Removed all manual armor handling (base64 decoding, BEGIN/END lines, Peek check) and now use `armor.NewReader` directly. The code is simpler and fully relies on the armor reader as you suggested. All tests pass. Please review when you have a chance. Thanks!

AnomalRoil commented on cmd/tle/commands/metadata.go:
```diff
@@ -0,0 +1,117 @@
+}
+
+// Metadata reads INPUT from src and, if it contains a tlock stanza, outputs YAML with round, chainhash and estimated time.
+func Metadata(dst io.Writer, src io.Reader, network *http.Network) error {
```
Could you add a test for that function?

Just take a random testdata like https://github.com/drand/tlock/blob/main/testdata/lorem-tle-testnet-quicknet-t-2024-01-17-15-28.tle and try to extract metadata.

alienx5499 commented:
Hi @AnomalRoil,
Added a test for the Metadata function using the testdata file lorem-tle-testnet-quicknet-t-2024-01-17-15-28.tle as requested. The test verifies metadata extraction (round, chain_hash, time) from armored age files. All tests pass.
Ready for review.

AnomalRoil commented:
Thanks a lot.
